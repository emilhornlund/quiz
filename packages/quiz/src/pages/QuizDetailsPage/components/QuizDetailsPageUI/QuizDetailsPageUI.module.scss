@use '../../../../styles/colors';
@use '../../../../styles/helpers';
@use '../../../../styles/variables';

.details {
  display: grid;
  border-style: solid;
  border-color: colors.$white-1-50;
  background-color: colors.$white-1-80;

  @include helpers.mobile {
    grid-template-columns: 1fr 1fr;
    gap: variables.$mobile-narrow-layout-gutter;
    width: 100%;
    border-width: helpers.pxToRem(2px);
    border-radius: variables.$mobile-narrow-layout-gutter;
    padding: variables.$mobile-narrow-layout-gutter;
  }

  @include helpers.tablet {
    grid-template-columns: 1fr 1fr 1fr;
    gap: variables.$tablet-narrow-layout-gutter;
    width: variables.$tablet-width;
    border-width: helpers.pxToRem(3px);
    border-radius: variables.$tablet-narrow-layout-gutter;
    padding: variables.$tablet-narrow-layout-gutter;
  }

  @include helpers.desktop {
    grid-template-columns: 1fr 1fr 1fr 1fr;
    gap: variables.$desktop-narrow-layout-gutter;
    width: variables.$tablet-width;
    border-width: helpers.pxToRem(4px);
    border-radius: variables.$desktop-narrow-layout-gutter;
    padding: variables.$desktop-narrow-layout-gutter;
  }

  .item {
    display: flex;
    flex-direction: column;
    color: colors.$gray-4;
    text-align: center;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;

    @include helpers.mobile {
      gap: calc(#{variables.$mobile-narrow-layout-gutter} * 0.5);

      @include helpers.fontSize(16px);
    }

    @include helpers.tablet {
      gap: calc(#{variables.$tablet-narrow-layout-gutter} * 0.5);

      @include helpers.fontSize(20px);
    }

    @include helpers.desktop {
      gap: calc(#{variables.$desktop-narrow-layout-gutter} * 0.5);

      @include helpers.fontSize(18px);
    }
  }
}
