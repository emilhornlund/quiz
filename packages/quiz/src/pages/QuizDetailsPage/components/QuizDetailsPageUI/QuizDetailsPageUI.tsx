import {
  faCalendarCheck,
  faCalendarPlus,
  faCircleQuestion,
  faEye,
  faGamepad,
  faIcons,
  faLanguage,
  faPen,
  faPlay,
  faTrash,
  faUser,
} from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { QuizResponseDto } from '@quiz/common'
import { format } from 'date-fns'
import { toZonedTime } from 'date-fns-tz'
import React, { FC, useState } from 'react'

import {
  Button,
  ConfirmDialog,
  LoadingSpinner,
  Page,
  ResponsiveImage,
  Typography,
} from '../../../../components'
import {
  GameModeLabels,
  LanguageLabels,
  QuizCategoryLabels,
  QuizVisibilityLabels,
} from '../../../../models'

import styles from './QuizDetailsPageUI.module.scss'

export interface QuizDetailsPageUIProps {
  quiz?: QuizResponseDto
  isOwner?: boolean
  isLoadingQuiz?: boolean
  isHostGameLoading?: boolean
  isDeleteQuizLoading?: boolean
  onHostGame: () => void
  onEditQuiz: () => void
  onDeleteQuiz: () => void
}

const QuizDetailsPageUI: FC<QuizDetailsPageUIProps> = ({
  quiz,
  isOwner = false,
  isLoadingQuiz,
  isHostGameLoading,
  isDeleteQuizLoading,
  onHostGame,
  onEditQuiz,
  onDeleteQuiz,
}) => {
  const [showConfirmDeleteDialog, setShowConfirmDeleteDialog] = useState(false)

  const [showConfirmHostGameModal, setShowConfirmHostGameModal] =
    useState<boolean>(false)

  if (!quiz || isLoadingQuiz) {
    return (
      <Page align="start" height="full" profile>
        <LoadingSpinner />
      </Page>
    )
  }

  return (
    <Page
      align="start"
      height="full"
      header={
        <>
          {isOwner && (
            <>
              <Button
                id="delete-quiz-button"
                type="button"
                kind="destructive"
                size="small"
                value="Delete"
                hideValue="mobile"
                icon={faTrash}
                onClick={() => setShowConfirmDeleteDialog(true)}
              />
              <Button
                id="edit-quiz-button"
                type="button"
                kind="primary"
                size="small"
                value="Edit"
                hideValue="mobile"
                icon={faPen}
                onClick={onEditQuiz}
              />
            </>
          )}
          <Button
            id="host-game-button"
            type="button"
            kind="call-to-action"
            size="small"
            value="Host Game"
            hideValue="mobile"
            icon={faPlay}
            onClick={() => setShowConfirmHostGameModal(true)}
          />
        </>
      }
      discover
      profile>
      <Typography variant="subtitle" size="medium">
        {quiz.title}
      </Typography>

      {quiz.description && (
        <Typography variant="text" size="medium">
          {quiz.description}
        </Typography>
      )}

      {quiz.imageCoverURL && <ResponsiveImage imageURL={quiz.imageCoverURL} />}

      <div className={styles.details}>
        <div className={styles.item}>
          <FontAwesomeIcon icon={faEye} />
          {QuizVisibilityLabels[quiz.visibility]}
        </div>
        <div className={styles.item}>
          <FontAwesomeIcon icon={faIcons} />
          {QuizCategoryLabels[quiz.category]}
        </div>
        <div className={styles.item}>
          <FontAwesomeIcon icon={faLanguage} />
          {LanguageLabels[quiz.languageCode]}
        </div>
        <div className={styles.item}>
          <FontAwesomeIcon icon={faGamepad} />
          {GameModeLabels[quiz.mode]}
        </div>
        <div className={styles.item}>
          <FontAwesomeIcon icon={faCircleQuestion} />
          {quiz.numberOfQuestions}{' '}
          {quiz.numberOfQuestions === 1 ? 'Question' : 'Questions'}
        </div>
        <div
          className={styles.item}
          title={`Created ${format(toZonedTime(quiz.created, 'UTC'), 'y-LL-dd HH:mm:ss')}`}>
          <FontAwesomeIcon icon={faCalendarPlus} />
          {format(toZonedTime(quiz.created, 'UTC'), 'y-LL-dd HH:mm')}
        </div>
        <div
          className={styles.item}
          title={`Updated ${format(toZonedTime(quiz.updated, 'UTC'), 'y-LL-dd HH:mm:ss')}`}>
          <FontAwesomeIcon icon={faCalendarCheck} />
          {format(toZonedTime(quiz.updated, 'UTC'), 'y-LL-dd HH:mm')}
        </div>
        <div className={styles.item}>
          <FontAwesomeIcon icon={faUser} />
          {quiz.author.name || 'N/A'}
        </div>
      </div>

      <ConfirmDialog
        title="Host Game"
        message="Are you sure you want to start hosting a new game? Players will be able to join as soon as the game starts."
        open={showConfirmHostGameModal}
        loading={isHostGameLoading}
        onConfirm={onHostGame}
        onClose={() => setShowConfirmHostGameModal(false)}
      />

      <ConfirmDialog
        title="Delete Quiz"
        message="Are you sure you want to delete this quiz?"
        open={showConfirmDeleteDialog}
        loading={isDeleteQuizLoading}
        onConfirm={onDeleteQuiz}
        onClose={() => setShowConfirmDeleteDialog(false)}
        destructive
      />
    </Page>
  )
}

export default QuizDetailsPageUI
