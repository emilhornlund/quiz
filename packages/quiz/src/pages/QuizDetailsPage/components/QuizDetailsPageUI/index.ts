export type { QuizDetailsPageUIProps } from './QuizDetailsPageUI'
export { default } from './QuizDetailsPageUI'
