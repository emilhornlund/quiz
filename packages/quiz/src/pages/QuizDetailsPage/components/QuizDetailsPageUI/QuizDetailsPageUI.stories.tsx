import {
  GameMode,
  LanguageCode,
  QuizCategory,
  QuizVisibility,
} from '@quiz/common'
import type { Meta, StoryObj } from '@storybook/react'
import { withRouter } from 'storybook-addon-remix-react-router'
import { v4 as uuidv4 } from 'uuid'

import QuizDetailsPageUI from './QuizDetailsPageUI'

const meta = {
  title: 'Pages/QuizDetailsPageUI',
  component: QuizDetailsPageUI,
  decorators: [withRouter],
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof QuizDetailsPageUI>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    quiz: {
      id: uuidv4(),
      title: 'The Ultimate Geography Challenge',
      description:
        'Test your knowledge of world capitals, landmarks, and continents in this fun and educational geography quiz.',
      mode: GameMode.Classic,
      visibility: QuizVisibility.Public,
      category: QuizCategory.GeneralKnowledge,
      imageCoverURL:
        'https://0utwqfl7.cdn.imgeng.in/explore-academics/programs/images/undergraduate/henson/geographymajorMH.jpg',
      languageCode: LanguageCode.English,
      numberOfQuestions: 14,
      author: { id: uuidv4(), name: 'FrostyBear' },
      created: new Date(),
      updated: new Date(),
    },
    isLoadingQuiz: false,
    isHostGameLoading: false,
    isDeleteQuizLoading: false,
    onHostGame: () => undefined,
    onEditQuiz: () => undefined,
    onDeleteQuiz: () => undefined,
  },
} satisfies Story
