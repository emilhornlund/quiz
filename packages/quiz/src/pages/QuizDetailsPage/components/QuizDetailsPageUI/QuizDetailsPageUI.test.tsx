import {
  GameMode,
  LanguageCode,
  QuizCategory,
  QuizVisibility,
} from '@quiz/common'
import { render } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { describe, expect, it } from 'vitest'

import QuizDetailsPageUI from './QuizDetailsPageUI'

const created = new Date('2025-02-14T15:31:14.000Z')
const updated = new Date('2025-03-08T15:31:14.000Z')

describe('QuizDetailsPageUI', () => {
  it('should render QuizDetailsPageUI', async () => {
    const { container } = render(
      <MemoryRouter>
        <QuizDetailsPageUI
          quiz={{
            id: 'd12cf443-3fa9-4a0e-8778-d9c182903146',
            title: 'The Ultimate Geography Challenge',
            description:
              'Test your knowledge of world capitals, landmarks, and continents in this fun and educational geography quiz.',
            mode: GameMode.Classic,
            visibility: QuizVisibility.Public,
            category: QuizCategory.GeneralKnowledge,
            imageCoverURL:
              'https://0utwqfl7.cdn.imgeng.in/explore-academics/programs/images/undergraduate/henson/geographymajorMH.jpg',
            languageCode: LanguageCode.English,
            numberOfQuestions: 14,
            author: {
              id: 'db8d4c90-bfc2-4c2e-93cc-8f1c7eda34ec',
              name: 'FrostyBear',
            },
            created,
            updated,
          }}
          isLoadingQuiz={false}
          isHostGameLoading={false}
          isDeleteQuizLoading={false}
          onHostGame={() => undefined}
          onEditQuiz={() => undefined}
          onDeleteQuiz={() => undefined}
        />
      </MemoryRouter>,
    )

    expect(container).toMatchSnapshot()
  })
})
