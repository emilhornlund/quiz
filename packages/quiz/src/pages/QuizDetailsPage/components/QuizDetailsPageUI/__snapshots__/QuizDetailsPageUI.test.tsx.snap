// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QuizDetailsPageUI > should render QuizDetailsPageUI 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <a
          data-discover="true"
          href="/auth/login"
        >
          Login
        </a>
        <div
          class="buttonInputContainer buttonInputKindCallToAction buttonInputSizeSmall"
        >
          <button
            data-testid="test-host-game-button-button"
            id="host-game-button"
            name="host-game-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-play "
              data-icon="play"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80L0 432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
                fill="currentColor"
              />
            </svg>
            <span>
              Host Game
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="content fullWidth fullHeight startAlign"
    >
      <div
        class="typography subtitle medium"
      >
        The Ultimate Geography Challenge
      </div>
      <div
        class="typography text medium"
      >
        Test your knowledge of world capitals, landmarks, and continents in this fun and educational geography quiz.
      </div>
      <div
        class="container"
      >
        <img
          class="image"
          src="https://0utwqfl7.cdn.imgeng.in/explore-academics/programs/images/undergraduate/henson/geographymajorMH.jpg"
        />
      </div>
      <div
        class="details"
      >
        <div
          class="item"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye "
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288 32c-80.8 0-145.5 36.8-192.6 80.6C48.6 156 17.3 208 2.5 243.7c-3.3 7.9-3.3 16.7 0 24.6C17.3 304 48.6 356 95.4 399.4C142.5 443.2 207.2 480 288 480s145.5-36.8 192.6-80.6c46.8-43.5 78.1-95.4 93-131.1c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C433.5 68.8 368.8 32 288 32zM144 256a144 144 0 1 1 288 0 144 144 0 1 1 -288 0zm144-64c0 35.3-28.7 64-64 64c-7.1 0-13.9-1.2-20.3-3.3c-5.5-1.8-11.9 1.6-11.7 7.4c.3 6.9 1.3 13.8 3.2 20.7c13.7 51.2 66.4 81.6 117.6 67.9s81.6-66.4 67.9-117.6c-11.1-41.5-47.8-69.4-88.6-71.1c-5.8-.2-9.2 6.1-7.4 11.7c2.1 6.4 3.3 13.2 3.3 20.3z"
              fill="currentColor"
            />
          </svg>
          Public
        </div>
        <div
          class="item"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-icons "
            data-icon="icons"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M500.3 7.3C507.7 13.3 512 22.4 512 32l0 144c0 26.5-28.7 48-64 48s-64-21.5-64-48s28.7-48 64-48l0-57L352 90.2 352 208c0 26.5-28.7 48-64 48s-64-21.5-64-48s28.7-48 64-48l0-96c0-15.3 10.8-28.4 25.7-31.4l160-32c9.4-1.9 19.1 .6 26.6 6.6zM74.7 304l11.8-17.8c5.9-8.9 15.9-14.2 26.6-14.2l61.7 0c10.7 0 20.7 5.3 26.6 14.2L213.3 304l26.7 0c26.5 0 48 21.5 48 48l0 112c0 26.5-21.5 48-48 48L48 512c-26.5 0-48-21.5-48-48L0 352c0-26.5 21.5-48 48-48l26.7 0zM192 408a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM478.7 278.3L440.3 368l55.7 0c6.7 0 12.6 4.1 15 10.4s.6 13.3-4.4 17.7l-128 112c-5.6 4.9-13.9 5.3-19.9 .9s-8.2-12.4-5.3-19.2L391.7 400 336 400c-6.7 0-12.6-4.1-15-10.4s-.6-13.3 4.4-17.7l128-112c5.6-4.9 13.9-5.3 19.9-.9s8.2 12.4 5.3 19.2zm-339-59.2c-6.5 6.5-17 6.5-23 0L19.9 119.2c-28-29-26.5-76.9 5-103.9c27-23.5 68.4-19 93.4 6.5l10 10.5 9.5-10.5c25-25.5 65.9-30 93.9-6.5c31 27 32.5 74.9 4.5 103.9l-96.4 99.9z"
              fill="currentColor"
            />
          </svg>
          General Knowledge
        </div>
        <div
          class="item"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-language "
            data-icon="language"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64l192 0 48 0 16 0 256 0c35.3 0 64 28.7 64 64l0 256c0 35.3-28.7 64-64 64l-256 0-16 0-48 0L64 448c-35.3 0-64-28.7-64-64L0 128zm320 0l0 256 256 0 0-256-256 0zM178.3 175.9c-3.2-7.2-10.4-11.9-18.3-11.9s-15.1 4.7-18.3 11.9l-64 144c-4.5 10.1 .1 21.9 10.2 26.4s21.9-.1 26.4-10.2l8.9-20.1 73.6 0 8.9 20.1c4.5 10.1 16.3 14.6 26.4 10.2s14.6-16.3 10.2-26.4l-64-144zM160 233.2L179 276l-38 0 19-42.8zM448 164c11 0 20 9 20 20l0 4 44 0 16 0c11 0 20 9 20 20s-9 20-20 20l-2 0-1.6 4.5c-8.9 24.4-22.4 46.6-39.6 65.4c.9 .6 1.8 1.1 2.7 1.6l18.9 11.3c9.5 5.7 12.5 18 6.9 27.4s-18 12.5-27.4 6.9l-18.9-11.3c-4.5-2.7-8.8-5.5-13.1-8.5c-10.6 7.5-21.9 14-34 19.4l-3.6 1.6c-10.1 4.5-21.9-.1-26.4-10.2s.1-21.9 10.2-26.4l3.6-1.6c6.4-2.9 12.6-6.1 18.5-9.8l-12.2-12.2c-7.8-7.8-7.8-20.5 0-28.3s20.5-7.8 28.3 0l14.6 14.6 .5 .5c12.4-13.1 22.5-28.3 29.8-45L448 228l-72 0c-11 0-20-9-20-20s9-20 20-20l52 0 0-4c0-11 9-20 20-20z"
              fill="currentColor"
            />
          </svg>
          English
        </div>
        <div
          class="item"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-gamepad "
            data-icon="gamepad"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M192 64C86 64 0 150 0 256S86 448 192 448l256 0c106 0 192-86 192-192s-86-192-192-192L192 64zM496 168a40 40 0 1 1 0 80 40 40 0 1 1 0-80zM392 304a40 40 0 1 1 80 0 40 40 0 1 1 -80 0zM168 200c0-13.3 10.7-24 24-24s24 10.7 24 24l0 32 32 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-32 0 0 32c0 13.3-10.7 24-24 24s-24-10.7-24-24l0-32-32 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l32 0 0-32z"
              fill="currentColor"
            />
          </svg>
          Classic
        </div>
        <div
          class="item"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-question "
            data-icon="circle-question"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
              fill="currentColor"
            />
          </svg>
          14
           
          Questions
        </div>
        <div
          class="item"
          title="Created 2025-02-14 15:31:14"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-calendar-plus "
            data-icon="calendar-plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 32l0 32L48 64C21.5 64 0 85.5 0 112l0 48 448 0 0-48c0-26.5-21.5-48-48-48l-48 0 0-32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 32L160 64l0-32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192L0 192 0 464c0 26.5 21.5 48 48 48l352 0c26.5 0 48-21.5 48-48l0-272zM224 248c13.3 0 24 10.7 24 24l0 56 56 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-56 0 0 56c0 13.3-10.7 24-24 24s-24-10.7-24-24l0-56-56 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l56 0 0-56c0-13.3 10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
          2025-02-14 15:31
        </div>
        <div
          class="item"
          title="Updated 2025-03-08 15:31:14"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-calendar-check "
            data-icon="calendar-check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M128 0c17.7 0 32 14.3 32 32l0 32 128 0 0-32c0-17.7 14.3-32 32-32s32 14.3 32 32l0 32 48 0c26.5 0 48 21.5 48 48l0 48L0 160l0-48C0 85.5 21.5 64 48 64l48 0 0-32c0-17.7 14.3-32 32-32zM0 192l448 0 0 272c0 26.5-21.5 48-48 48L48 512c-26.5 0-48-21.5-48-48L0 192zM329 305c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-95 95-47-47c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9l64 64c9.4 9.4 24.6 9.4 33.9 0L329 305z"
              fill="currentColor"
            />
          </svg>
          2025-03-08 15:31
        </div>
        <div
          class="item"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user "
            data-icon="user"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512l388.6 0c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304l-91.4 0z"
              fill="currentColor"
            />
          </svg>
          FrostyBear
        </div>
      </div>
    </div>
  </div>
</div>
`;
